build.gradle.kts

```diff
@@ -10,86 +9,115 @@ description = "A Result monad for modelling success or failure operations."
-    java {
-        srcDir("src/benchmarks/kotlin")
-    }
-}
-
-val benchmarkImplementation by configurations
+// creating this sourceSet and specifying jvm { withJava() } will create a
+// "jvmBenchmark" sourceSet we can access from kotlin mpp extension
+sourceSets.create("benchmark")
 
 benchmark {
     targets {
-        register("benchmark")
+        register("jvmBenchmark")
     }
 }
 
-dependencies {
-    implementation(kotlin("stdlib"))
-    testImplementation("junit:junit:4.13")
-    testImplementation(kotlin("test-common"))
-    testImplementation(kotlin("test-annotations-common"))
-    testImplementation(kotlin("test-junit"))
-    testImplementation(kotlin("test"))
+kotlin {
+    jvm {
+        withJava()
+    }
+    sourceSets {
+        val commonMain by getting {
+            dependencies {
+                implementation(kotlin("stdlib-common"))
+            }
+        }
 
-    benchmarkImplementation(sourceSets.main.get().output + sourceSets.main.get().runtimeClasspath)
-    benchmarkImplementation("org.jetbrains.kotlinx:kotlinx.benchmark.runtime-jvm:0.2.0-dev-8")
-}
+        val commonTest by getting {
+            dependencies {
+                implementation(kotlin("test-common"))
+                implementation(kotlin("test-annotations-common"))
+            }
+        }
 
-tasks.withType<DependencyUpdatesTask> {
-    rejectVersionIf {
-        listOf("alpha", "beta", "rc", "cr", "m", "eap", "pr").any {
-            candidate.version.contains(it, ignoreCase = true)
+        val jvmMain by getting {
+            dependencies {
+                implementation(kotlin("stdlib-jdk8"))
+            }
+        }
+
+        val jvmTest by getting {
+            dependencies {
+                implementation(kotlin("test-junit"))
+                implementation(kotlin("test"))
+            }
+        }
+
+        val jvmBenchmark by getting {
+            dependsOn(jvmMain)
+            dependencies {
+                implementation("org.jetbrains.kotlinx:kotlinx.benchmark.runtime-jvm:0.2.0-dev-8")
+            }
         }
+
     }
-}
 
-val dokka by tasks.existing(DokkaTask::class) {
-    outputFormat = "javadoc"
-    outputDirectory = "$buildDir/docs/javadoc"
-}
+    jvm().compilations.all {
+        kotlinOptions {
+            jvmTarget = "1.8"
+        }
+    }
 
-val javadocJar by tasks.registering(Jar::class) {
-    group = LifecycleBasePlugin.BUILD_GROUP
-    description = "Assembles a jar archive containing the Javadoc API documentation."
-    archiveClassifier.set("javadoc")
-    dependsOn(dokka)
-    from(dokka.get().outputDirectory)
-}
+    jvm {
+        mavenPublication {
+            artifact(javadocJar.get())
+        }
+    }
 
-val sourcesJar by tasks.registering(Jar::class) {
-    group = LifecycleBasePlugin.BUILD_GROUP
-    description = "Assembles a jar archive containing the main classes with sources."
-    archiveClassifier.set("sources")
-    from(sourceSets["main"].allSource)
+    sourceSets.all {
```

Comment:
Surely this can be put inside the other `sourceSets` block?

```kotlin
sourceSets {
    all {
         ...
    }

    named("commonMain") {
        ...
    }

...
...
```

Reply:
makes sense! 👍

Reply:
done